crates/ruff_python_formatter/README.md

```diff
@@ -94,18 +94,29 @@ write!(f, [if_group_breaks(&text(","))])
 If you need avoid second mutable borrows with a builder, you can use `format_with(|f| { ... })` as
 a formattable element similar to `text()` or `group()`.
 
-The generic comment formatting in `FormatNodeRule` handles comments correctly for most nodes, e.g.
-preceding and end-of-line comments depending on the node range. Sometimes however, you may have
-dangling comments that are not before or after a node but inside of it, e.g.
+## Comments
+
+Comments can either be own line or end-of-line and can be marked as `Leading`, `Trailing` and `Dangling`.
+
+```python
+# Leading comment (always own line)
+print("hello world")  # Trailing comment (end-of-line)
+# Trailing comment (own line)
+```
+
+Comment are automatically attached as `Leading` or `Trailing` to a node close to them categorization
+is automatic except when overridden in `place_comment` in `placement.rs`, which this section is
+about. A `Dangling` comment happens when there is node that the comment would be leading or trailing
+to.
 
 ```Python
 [
     # here we use an empty list
 ]
 ```
 
-Here, you have to call `dangling_comments` manually and stubbing out `fmt_dangling_comments` in list
-formatting.
+Here, you have to call `dangling_comments` manually and stubbing out `fmt_dangling_comments` default
```

Comment:
if the comment inside the empty list is an example of a dangling comment, can we make it say something about that instead of talking about the python code?